cmd/snap-confine/mount-support.c

```diff
@@ -301,7 +305,13 @@ static void sc_bootstrap_mount_namespace(const struct sc_mount_config *config)
 		}
 		sc_must_snprintf(dst, sizeof dst, "%s/%s", scratch_dir,
 				 mnt->path);
-		sc_do_mount(mnt->path, dst, NULL, MS_REC | MS_BIND, NULL);
+		if (mnt->is_optional) {
+			sc_do_optional_mount(mnt->path, dst, NULL,
+					     MS_REC | MS_BIND, NULL);
+		} else {
+			sc_do_mount(mnt->path, dst, NULL, MS_REC | MS_BIND,
+				    NULL);
+		}
 		if (!mnt->is_bidirectional) {
```

Comment:
How does optional and bidirectional fit together?

Reply:
It didn't, good catch! I fixed the logic to skip any sharing operation changes (bidirectionality) if mount itself has failed.